helm/templates/daemonset.yaml

```diff
@@ -28,36 +30,40 @@ spec:
             - {{ $k }}.efs.{{ $v.region }}.amazonaws.com
       {{- end }}
     {{- end }}
-    {{- if .Values.imagePullSecrets }}
+    {{- with .Values.imagePullSecrets }}
       imagePullSecrets:
-      {{- range .Values.imagePullSecrets }}
-        - name: {{ . }}
-      {{- end }}
+        {{- toYaml . | nindent 8 }}
     {{- end }}
+      serviceAccountName: {{ template "aws-efs-csi-driver.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- with .Values.nodeSelector }}
       nodeSelector:
-        beta.kubernetes.io/os: linux
```

Comment:
Is it intentional to lose this selector?

Reply:
No, but I would like to move it into values.yaml to give the user more choice. Will fix.